### ISSUE TITLE
Questions about the testing process

### ISSUE THREAD

**Issue by licong-123** [Hxyz-123/Font-diff#4]
Hi, I have encountered some problems during the testing process. I thought the generated image would be the Chinese characters in gen_char.txt, but the actual result is some irrelevant Chinese characters and the generated result is different for each test. Is my understanding incorrect and how do I generate specific characters?
### test_cfg.yaml
`
dropout: 0.1
chara_nums: 6625
diffusion_steps: 1000
noise_schedule: 'linear'
image_size: 80
num_channels: 128
num_res_blocks: 3
batch_size: 5
num_samples: 10
attention_resolutions: '40, 20, 10'
use_ddim: True
timestep_respacing: ddim25
stroke_path: './chinese_stroke.txt'
model_path: './trained_models/model800000.pt'
sty_img_path: './data/id_4/00001.png'
total_txt_file: './total_chn.txt'
gen_txt_file: './gen_char.txt'
img_save_path: './result'
classifier_free: True
cont_scale: 3.0
sk_scale: 3.0
`
### gen_char.txt
已通并提直题党程展五果

###generated result
![image](https://user-images.githubusercontent.com/132633816/236657444-dfe19191-8c8c-40ec-85bc-5dd1a406e9d3.png)

**Comment by chenjingcheng** [comment]
I meet the same problem with you.

**Comment by licong-123** [comment]
> I meet the same problem with you.

Hi, do you know how to solve this problem yet?

**Comment by chenjingcheng** [comment]
> > I meet the same problem with you.
> 
> Hi, do you know how to solve this problem yet?

sorry,No

**Comment by Hxyz-123** [comment]
> > I meet the same problem with you.
> 
> Hi, do you know how to solve this problem yet?

Hello, can you tell me how you create your own dataset?

**Comment by licong-123** [comment]
> > > I meet the same problem with you.
> > 
> > 
> > Hi, do you know how to solve this problem yet?
> 
> Hello, can you tell me how you create your own dataset?

Hi, the calligraphy data I used is some authentic calligraphy images that I cropped. I first generated the kai images using the ttf file and font2img.py. For the calligraphy images I first found the location of the character in total_chn.txt and then renamed the images according to the location of the character. Is this correct, please?
### Example of a calligraphy dataset
![image](https://github.com/Hxyz-123/Font-diff/assets/132633816/d06523b7-40c2-465d-bf2f-75abb9e3b63a)

**Comment by Hxyz-123** [comment]
> > > > I meet the same problem with you.
> > > 
> > > 
> > > Hi, do you know how to solve this problem yet?
> > 
> > 
> > Hello, can you tell me how you create your own dataset?
> 
> Hi, the calligraphy data I used is some authentic calligraphy images that I cropped. I first generated the kai images using the ttf file and font2img.py. For the calligraphy images I first found the location of the character in total_chn.txt and then renamed the images according to the location of the character. Is this correct, please?
> 
> ### Example of a calligraphy dataset
> ![image](https://user-images.githubusercontent.com/132633816/237614818-d06523b7-40c2-465d-bf2f-75abb9e3b63a.png)

Since we use token embeddings to represent the content, the training set should contain all characters images in 'total_chn.txt', or you can create your own 'total_chn.txt' and 'chinese_stroke.txt' and modify the 'chara_nums'.

**Comment by chenjingcheng** [comment]
Using font2img.py to generate PNG images without modifying the image data

**Comment by chenjingcheng** [comment]
Hello, you closed the bug. May I ask how you resolved it? Thank you!

**Comment by licong-123** [comment]
> Hello, you closed the bug. May I ask how you resolved it? Thank you!
I have not addressed this problem. This is because the author says that the training set should contain all the character images in 'total_chn.txt', which means that it should contain the same characters for each type of calligraphy data. But I used real calligraphy images, not ttf files, it was not very realistic for each font to contain the same Chinese characters, so I gave up using the model.
